aiplatform/src/main/java/aiplatform/EmbeddingModelTuningSample.java

```diff
@@ -0,0 +1,129 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package aiplatform;
+
+// [START aiplatform_sdk_embedding_model_tuning]
+import com.google.cloud.aiplatform.v1.CreatePipelineJobRequest;
+import com.google.cloud.aiplatform.v1.LocationName;
+import com.google.cloud.aiplatform.v1.PipelineJob;
+import com.google.cloud.aiplatform.v1.PipelineJob.RuntimeConfig;
+import com.google.cloud.aiplatform.v1.PipelineServiceClient;
+import com.google.cloud.aiplatform.v1.PipelineServiceSettings;
+import com.google.protobuf.Value;
+import java.io.IOException;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+// [END aiplatform_sdk_embedding_model_tuning]
+
+public class EmbeddingModelTuningSample {
+  public static void main(String[] args) throws IOException {
+    // [START aiplatform_sdk_embedding_model_tuning]
+    // TODO(developer): Replace these variables before running this sample.
+    String apiEndpoint = "us-central1-aiplatform.googleapis.com:443";
+    String project = "PROJECT";
+    String baseModelVersionId = "BASE_MODEL_VERSION_ID";
+    String taskType = "TASK_TYPE";
+    String pipelineJobDisplayName = "PIPELINE_JOB_DISPLAY_NAME";
+    String outputDir = "OUTPUT_DIR";
+    String queriesPath = "QUERIES";
+    String corpusPath = "CORPUS";
+    String trainLabelPath = "TRAIN_LABEL";
+    String testLabelPath = "TEST_LABEL";
+    int batchSize = 50;
+    int iterations = 300;
+
+    createEmbeddingModelTuningPipelineJob(
+        apiEndpoint,
+        project,
+        baseModelVersionId,
+        taskType,
+        pipelineJobDisplayName,
+        outputDir,
+        queriesPath,
+        corpusPath,
+        trainLabelPath,
+        testLabelPath,
+        batchSize,
+        iterations);
+    // [END aiplatform_sdk_embedding_model_tuning]
+  }
+
+  // [START aiplatform_sdk_embedding_model_tuning]
+  public static PipelineJob createEmbeddingModelTuningPipelineJob(
+      String apiEndpoint,
+      String project,
+      String baseModelVersionId,
+      String taskType,
+      String pipelineJobDisplayName,
+      String outputDir,
+      String queriesPath,
+      String corpusPath,
+      String trainLabelPath,
+      String testLabelPath,
+      int batchSize,
+      int iterations)
+      throws IOException {
+    Pattern apiEndpointPattern =
+        Pattern.compile("(?<Location>.+)(-autopush|-staging)?-aiplatform.+");
+    Matcher matcher = apiEndpointPattern.matcher(apiEndpoint);
```

Comment:
nit: do we really need it in the code sample? For example the code does not validate project ID and other input parameters.

Reply:
This is not optional, but rather required. This code extracts the required location from an API endpoint.